ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh

```diff
@@ -1,21 +1,67 @@
 #!/usr/bin/env bash
 
 set -x
+set +e
+
+BASE_DIR=${HOME:-"/alabama"}
+MUST_GATHER_TIMEOUT=${MUST_GATHER_TIMEOUT:-"5m"}
+NS_OPERATORS=${NS_OPERATORS:-"openstack"}
+NS_SERVICES=${NS_SERVICES:-"openstack"}
+
+function create_out_dir {
+  local OUT_DIR=${1}
+  if [ ! -d "./${OUT_DIR}" ]; then
+    mkdir ${OUT_DIR}
+  fi
+}
+
+function gen_pods_debug_cmd {
+  create_out_dir ${1}
+  local cmd=("echo 'Logs for pod: {}'; oc describe pod {} > ${1}/{}-describe.log; oc logs --all-containers=true {} >> ${1}/{}-containers.log")
+  echo "${cmd[@]}"
+}
+
+function gen_api_resources_debug_cmd {
+  create_out_dir ${1}
+  local cmd=("echo 'Info for api-resource: {}'; oc get {} -o yaml > ${1}/{}.yaml")
+  echo "${cmd[@]}"
+}
 
 # We don't want to use OpenShift-CI build cluster namespace
 unset NAMESPACE
+oc project "${NS_OPERATORS}"
 
-oc project openstack
+pushd "${BASE_DIR}" || exit
+mkdir logs
+pushd logs || exit
 
-oc logs ceph
+oc get all > get_all.log
 
-oc get OpenStackControlPlane openstack -o json
+CMD=$(gen_api_resources_debug_cmd "resources")
+oc api-resources | grep 'openstack' | awk '{print $1}' | xargs -n1 -I {} sh -c "${CMD[@]}"
+# RabbitMQCluster
+oc get -o yaml RabbitMQCluster > RabbitMQCluster.yaml
 
-oc get all
+CMD=$(gen_pods_debug_cmd "controller-manager")
+oc get pods -n "${NS_OPERATORS}" --show-labels | grep -i '.*control\-plane\=.*controller\-manager.*' | awk '{print $1}' | \
+xargs -n1 -I {} sh -c "${CMD[@]}"
 
-oc get -o yaml MariaDB,RabbitMQCluster,KeystoneAPI,PlacementAPI,Glance,Cinder,NeutronAPI,Nova
-
-oc get pods -n openstack --show-labels | grep -i '.*control\-plane\=.*controller\-manager.*' | awk '{print $1}' | xargs -n1 oc logs
+CMD=$(gen_pods_debug_cmd "services")
+oc get pods -n "${NS_SERVICES}" --no-headers | egrep -iv controller | awk '{print $1}' | xargs -n1 -I {} sh -c "${CMD[@]}"
 
+CMD=$(gen_pods_debug_cmd "errors")
 # Capture logs from service containers if container is not in Running or Completed state
-oc get pods -n openstack --no-headers | egrep -iv controller | egrep -iv 'Running|Completed' | awk '{print $1}' | xargs -n1 -I {} sh -c 'echo "Logs for pod: {}"; oc describe pod {}; oc logs --all-containers=true {}'
+oc get pods -n "${NS_SERVICES}" --no-headers | egrep -iv controller | egrep -iv 'Running|Completed' | awk '{print $1}' | \
+xargs -n1 -I {} sh -c "${CMD[@]}"
+
```

Comment:
- we should also collect details of  all  services, routes, deployments, daemonset, replicaset. 

could just loop over 
`oc get all -o name -n "${NS_SERVICES}" `
and then dump the yamls for each

- Also should collect all NADS of the namespace (only exist in network isolation)

`oc get network-attachment-definition -n "${NS_SERVICES}"`

- and the nncps (only exist in network isolation)
`oc get nncp`

- metallb resource (only exist in network isolation)
`oc get ipaddresspool -n metallb-system`
`oc get l2advertisement -n metallb-system`

Reply:
Thanks for the inputs, I will work on that too.

Reply:
Added, some outputs here[1]

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37084/rehearse-37084-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest/1636161182786129920/artifacts/openstack-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/